src/main/scala/wdl4s/WdlNamespace.scala

```diff
@@ -111,28 +112,24 @@ case class NamespaceWithWorkflow(importedAs: Option[String],
    * and return the value for storage in the symbol store
    */
   def staticDeclarationsRecursive(userInputs: WorkflowCoercedInputs, wdlFunctions: WdlStandardLibraryFunctions): Try[WorkflowCoercedInputs] = {
-    import scala.collection.mutable
-    val collected = mutable.Map[String, WdlValue]()
-    val allDeclarations = workflow.declarations ++ workflow.calls.flatMap {_.task.declarations}
-
-    val evaluatedDeclarations = allDeclarations.filter {_.expression.isDefined}.map {decl =>
-      val value = decl.expression.get.evaluate(declarationLookupFunction(decl, collected.toMap ++ userInputs), wdlFunctions)
-      collected += (decl.fullyQualifiedName -> value.get)
-      val coercedValue = value match {
-        case Success(s) => decl.wdlType.coerceRawValue(s)
-        case f => f
-      }
-      decl.fullyQualifiedName -> coercedValue
-    }.toMap
 
-    val (successes, failures) = evaluatedDeclarations.partition {case (_, tryValue) => tryValue.isSuccess}
-    if (failures.isEmpty) {
-      Success(successes.map {case (k,v) => k -> v.get})
-    } else {
-      val errors = failures.values.collect { case f: Failure[_] => f.exception.getMessage }
-        // .get because failures is guaranteed to be nonEmpty
-      Failure(new ValidationException("Could not evaluate some declaration expressions", errors.toList.toNel.get))
+    def evalDeclaration(accumulated: Map[String, Try[WdlValue]], current: ScopedDeclaration): Map[String, Try[WdlValue]] = {
+      current.expression match {
+        case Some(expr) =>
+          val successfulAccumulated = accumulated.collect({case (k, v) if v.isSuccess => k -> v.get })
+          val value = expr.evaluate(declarationLookupFunction(current, successfulAccumulated ++ userInputs), wdlFunctions)
+          accumulated + (current.fullyQualifiedName -> value)
+        case None => accumulated
+      }
     }
+
+    val workflowStaticDeclarations = workflow.scopedDeclarations.foldLeft(Map.empty[String, Try[WdlValue]])(evalDeclaration)
+
+    val callStaticDeclarations = workflow.calls flatMap { call =>
+      call.scopedDeclarations.foldLeft(Map.empty[String, Try[WdlValue]])(evalDeclaration)
+    } toMap
+
+    TryUtil.sequenceMap(workflowStaticDeclarations ++ callStaticDeclarations)
```

Comment:
As a general rule of thumb across all of our code we should be moving away for our homebrew scheme which was only used because we weren't already using validations. Now that we are, there's a far more idiomatic way to express accumulation of badness.